zrml/global-disputes/src/lib.rs

```diff
@@ -647,8 +785,88 @@ mod pallet {
+
+            ensure!(
+                <GlobalDisputesInfo<T>>::get(market_id).is_none(),
+                Error::<T>::GlobalDisputeAlreadyExists
+            );
+
+            ensure!(initial_items.len() >= 2, Error::<T>::AtLeastTwoOutcomesRequired);
+
+            for InitialItem { outcome, owner, amount } in initial_items {
+                ensure!(market.matches_outcome_report(outcome), Error::<T>::OutcomeMismatch);
+
+                match <Outcomes<T>>::get(market_id, outcome) {
+                    Some(mut outcome_info) => {
+                        let outcome_sum = outcome_info.outcome_sum.saturating_add(*amount);
+                        outcome_info.outcome_sum = outcome_sum;
+                        let mut owners = outcome_info
+                            .possession
+                            .get_shared_owners()
+                            .ok_or(Error::<T>::SharedPossessionRequired)?;
+                        owners.try_push(owner.clone()).map_err(|_| Error::<T>::MaxOwnersReached)?;
+                        outcome_info.possession = Possession::Shared { owners };
+                        Self::update_winner(market_id, outcome, outcome_info.clone());
+                        <Outcomes<T>>::insert(market_id, outcome, outcome_info);
+                    }
+                    None => {
+                        // adding one item to BoundedVec can not fail
+                        if let Ok(owners) = BoundedVec::try_from(vec![owner.clone()]) {
+                            let possession = Possession::Shared { owners };
+                            let outcome_info = OutcomeInfo { outcome_sum: *amount, possession };
+                            Self::update_winner(market_id, outcome, outcome_info.clone());
+                            <Outcomes<T>>::insert(market_id, outcome, outcome_info);
+                        } else {
+                            log::error!("Global Disputes: Could not construct a bounded vector.");
+                            debug_assert!(false);
+                        }
+                    }
+                }
+            }
+
+            let now = <frame_system::Pallet<T>>::block_number();
+            let add_outcome_end = now.saturating_add(T::AddOutcomePeriod::get());
+            let vote_end = add_outcome_end.saturating_add(T::GdVotingPeriod::get());
+            let ids_len = T::DisputeResolution::add_auto_resolve(market_id, vote_end)?;
+
+            <GlobalDisputesInfo<T>>::try_mutate(market_id, |gd_info| -> DispatchResult {
+                let mut raw_gd_info = gd_info.as_mut().ok_or(Error::<T>::GlobalDisputeNotFound)?;
+                raw_gd_info.status = GdStatus::Active { add_outcome_end, vote_end };
+                *gd_info = Some(raw_gd_info.clone());
+                Ok(())
+            })?;
+
+            Ok(ids_len)
+        }
+
+        fn destroy_global_dispute(market_id: &MarketIdOf<T>) -> Result<(), DispatchError> {
```

Comment:
For example, what if we have a finished global dispute, the outcomes are partially purged and then we destroy the dispute. Does everything work correctly then? Is that scenario tested?

Reply:
The destroyment of a global dispute is possible with `admin_destroy_market`. In this case the ones who added voting outcomes are refunded and the ones who locked tokens are able to get everything unlocked. 

Oh, inside the prediction markets pallet the check was present before calling `destroy_global_dispute` that the status of the global dispute has to be active for the actual call to the function.

```rust
            #[cfg(feature = "with-global-disputes")]
            if T::GlobalDisputes::is_active(&market_id) {
                T::GlobalDisputes::destroy_global_dispute(&market_id)?;
            }
```

But I modified the code to make this more fail safe inside the global dispute pallet.

```rust

        fn destroy_global_dispute(market_id: &MarketIdOf<T>) -> Result<(), DispatchError> {
            <GlobalDisputesInfo<T>>::try_mutate(market_id, |gd_info| {
                let mut raw_gd_info = gd_info.as_mut().ok_or(Error::<T>::GlobalDisputeNotFound)?;
                
                // in case the global dispute is already finished nothing needs to be done
                if let GdStatus::Active { add_outcome_end: _, vote_end } = raw_gd_info.status {
                    T::DisputeResolution::remove_auto_resolve(market_id, vote_end);

                    raw_gd_info.status = GdStatus::Destroyed;
                    *gd_info = Some(raw_gd_info.clone());
                }
                
                Ok(())
            })
        }
```

Reply:
That's fine, we just have to keep in mind that a destroyed market doesn't necessarily imply that the global dispute has status `Destroyed`.